File: 코딩 기초 트레이닝/배열 조각하기-p181893/배열 조각하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181893

function solution(arr, query) {
    query.forEach((q, index) => {
        if(index % 2 === 0) {
            arr = arr.filter((a, i) => {
                return i <= q;
            });
        } else {
            arr = arr.filter((a, i) => {
                return i >= q;
            })
        }
    })

    return arr;
}

File: 코딩 기초 트레이닝/배열 조각하기-p181893/배열 조각하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181893

def solution(arr, query):
    for index, q in enumerate(query):
        if index % 2 == 0:
            arr = [x for i, x in enumerate(arr) if i <= q]
        else:
            arr = [x for i, x in enumerate(arr) if i >= q]

    return arr
